File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLSendMessageRecordVideoAction.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLSendMessageRecordVideoAction extends TLAbsSendMessageAction {

    public static final int CONSTRUCTOR_ID = 0xa187d66f;

    private final String _constructor = "sendMessageRecordVideoAction#a187d66f";

    public TLSendMessageRecordVideoAction() {
    }

    @Override
    public String toString() {
        return _constructor;
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLSendMessageRecordVideoAction.kt
package com.github.badoualy.telegram.tl.api

/**
 * sendMessageRecordVideoAction#a187d66f
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLSendMessageRecordVideoAction : TLAbsSendMessageAction() {
    private val _constructor: String = "sendMessageRecordVideoAction#a187d66f"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLSendMessageRecordVideoAction) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xa187d66f.toInt()
    }
}
